### ISSUE TITLE
Weird NoClassDefFoundError

### ISSUE THREAD

**Issue by bneil** [Verizon/helm#2]
I started a scala worksheet in intellij as well as when I am working in the cli, for some reason

```
import helm.ConsulOp
import helm.http4s._
import org.http4s.Uri.uri
import org.http4s.client.blaze.PooledHttp1Client
import scalaz.concurrent.Task

val client = PooledHttp1Client()
val baseUrl = uri("http://127.0.0.1:8500")
val interpreter = new Http4sConsulClient(baseUrl, client)

val s: Task[Unit] = helm.run(interpreter, ConsulOp.set("testkey", "testvalue"))
val g: Task[Option[String]] = helm.run(interpreter, ConsulOp.get("testkey"))

s.attemptRun
```

results in

```
import helm.ConsulOp
import helm.http4s._
import org.http4s.Uri.uri
import org.http4s.client.blaze.PooledHttp1Client
import scalaz.concurrent.Task

java.lang.NoClassDefFoundError: Could not initialize class org.http4s.client.blaze.Http1Support$
	at org.http4s.client.blaze.PooledHttp1Client$.apply(test.sc:14)
	at io.test.testapi.A$A36$A$A36.client$lzycompute(test.sc:7)
	at io.test.testapi.A$A36$A$A36.client(test.sc:7)
	at io.test.testapi.A$A36$A$A36.get$$instance$$client(test.sc:7)
	at #worksheet#.#worksheet#(test.sc:30)
```

my current build.sbt looks like

```
organization := "io.test"
name := "test-api"
version := "0.0.1-SNAPSHOT"
scalaVersion := "2.11.11"

val Http4sVersion = "0.17.0-M2"

resolvers += "OSS Sonatype" at "https://repo1.maven.org/maven2/"

libraryDependencies ++= Seq(
 "org.http4s"      %% "http4s-blaze-server" % Http4sVersion,
 "org.http4s"      %% "http4s-circe"        % Http4sVersion,
 "org.http4s"      %% "http4s-dsl"          % Http4sVersion,
 "org.http4s"      %% "http4s-core"         % Http4sVersion,
 "ch.qos.logback"  %  "logback-classic"     % "1.2.1",
 "io.verizon.helm" %% "http4s"              % "1.4.78-scalaz-7.1"
)
```

Ive tried a few different implentations but I keep getting a weird NoClassDefFoundError

```
package io.test.testapi

import helm._
import io.circe._
import org.http4s._
import org.http4s.circe._
import org.http4s.dsl._
import helm.http4s._
import org.http4s.Uri.uri
import org.http4s.client.blaze.PooledHttp1Client

import scalaz.concurrent.Task

object HelloWorld {
  lazy val client = PooledHttp1Client()
  val baseUrl = uri("http://172.17.0.2:8500")
  lazy val interpreter = new Http4sConsulClient(baseUrl, client)

  lazy val service = HttpService {
    case GET -> Root / "hello" / name =>
      val s: Task[Unit] = helm.run(interpreter, ConsulOp.set("testkey", "testvalue"))
      val g: Task[Option[String]] = helm.run(interpreter, ConsulOp.get("testkey"))

      // actually execute the calls
      s.run
      g.run
      Ok(Json.obj("message" -> Json.fromString(s"Hello, $name")))
  }
}
```
results in this:

```
java.lang.NoClassDefFoundError: org/http4s/util/CaseInsensitiveStringOps$
        at org.http4s.client.blaze.Http1Support$.<init>(Http1Support.scala:31)
        at org.http4s.client.blaze.Http1Support$.<clinit>(Http1Support.scala)
        at org.http4s.client.blaze.PooledHttp1Client$.apply(PooledHttp1Client.scala:18)
        at io.test.testapi.HelloWorld$.client$lzycompute(HelloWorld.scala:15)
        at io.test.testapi.HelloWorld$.client(HelloWorld.scala:15)
        at io.test.testapi.HelloWorld$.interpreter$lzycompute(HelloWorld.scala:17)
        at io.test.testapi.HelloWorld$.interpreter(HelloWorld.scala:17)
        at io.test.testapi.HelloWorld$$anonfun$service$1.applyOrElse(HelloWorld.scala:21)
        at io.test.testapi.HelloWorld$$anonfun$service$1.applyOrElse(HelloWorld.scala:19)
        at org.http4s.package$HttpService$$anonfun$apply$1.apply(package.scala:55)
        at org.http4s.package$HttpService$$anonfun$apply$1.apply(package.scala:55)
        at cats.data.KleisliSemigroup$$anonfun$combine$1.apply(Kleisli.scala:254)
        at org.http4s.server.blaze.Http1ServerStage.liftedTree1$1(Http1ServerStage.scala:112)
        at org.http4s.server.blaze.Http1ServerStage.runRequest(Http1ServerStage.scala:112)
        at org.http4s.server.blaze.Http1ServerStage.org$http4s$server$blaze$Http1ServerStage$$reqLoopCallback(Http1ServerStage.scala:95)
        at org.http4s.server.blaze.Http1ServerStage$$anonfun$requestLoop$1.apply(Http1ServerStage.scala:72)
        at org.http4s.server.blaze.Http1ServerStage$$anonfun$requestLoop$1.apply(Http1ServerStage.scala:72)
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:36)
        at org.http4s.blaze.util.Execution$$anon$1.execute(Executor.scala:27)
        at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:44)
        at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:252)
        at org.http4s.blaze.channel.nio1.NIO1HeadStage.readReady(NIO1HeadStage.scala:61)
        at org.http4s.blaze.channel.nio1.SelectorLoop.run(SelectorLoop.scala:130)
Caused by: java.lang.ClassNotFoundException: org.http4s.util.CaseInsensitiveStringOps$
        at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 23 common frames omitted
```

Usually id say I didnt have core on the classpath but ive verified it multiple times based on my build.sbt. Just curious if anyone has seen a behavior similar to this.

**Comment by rossabaker** [comment]
helm is currently built against http4s-0.15.  If you use http4s-0.15.11, these problems should go away.

When http4s-0.16 is released, a new version of helm will be published for scalaz 7.1 and 7.2 on that.

When http4s-0.17 is published, we'll have some thinking to do because helm is currently based on scalaz and http4s-0.17 is based on cats.

**Comment by bneil** [comment]
Understood, thanks for taking a look
